### PR TITLE
Fixed a seed value to reduce randomness for PyTorch models

### PR DESCRIPTION
**Pull request by karan6181**
- Fixed a seed value to reduce randomness for Benchmarking system

### CODE REVIEW

File: PyTorch/Segmentation/MaskRCNN/pytorch/train_net_herring.py [hunk 1]
```diff
@@ -232,9 +234,20 @@ def main():
...
+    parser.add_argument('--seed',
+                        '-s',
+                        help='manually set random seed for torch',
+                        type=int,
+                        default=99)
 
     args = parser.parse_args()
 
+    # Set seed to reduce randomness
+    random.seed(args.seed + herring.get_local_rank())
```
Comment (Cpruce) [inline PyTorch/Segmentation/MaskRCNN/pytorch/train_net_herring.py:246]: I think just setting the seeds to `args.seed` might be better since the calling script can use `OMPI_COMM_WORLD_LOCAL_RANK`

File: TensorFlow2/Segmentation/MaskRCNN/mask_rcnn_main.py [hunk 2]
```diff
@@ -111,6 +112,10 @@ def main(argv):
     dllogger.init(backends=[dllogger.JSONStreamBackend(verbosity=dllogger.Verbosity.VERBOSE,
                                                            filename=RUN_CONFIG.log_path)])
 
+    # Set seed to reduce randomness
+    np.random.seed(FLAGS.seed)
+    tf.set_random_seed(RUN_CONFIG.seed)
```
Comment (Cpruce) [inline TensorFlow2/Segmentation/MaskRCNN/mask_rcnn_main.py:117]: Can you also add using the determinism flags and container to your tests? We should see if there are any sources of non-determinism that setting the seeds does not resolve.

File: PyTorch/Segmentation/MaskRCNN/pytorch/train_net_herring.py (hunk 1 above)
Comment (karan6181) [reply PyTorch/Segmentation/MaskRCNN/pytorch/train_net_herring.py:246]: The PT BERT script used by NVIDIA does the similar thing: https://github.com/HerringForks/DeepLearningExamples/blob/master/PyTorch/LanguageModeling/BERT/run_pretraining.py#L499 .

So are you suggesting `random.seed(args.seed)` ?

Comment (piyushghai) [reply PyTorch/Segmentation/MaskRCNN/pytorch/train_net_herring.py:246]: Should the scripts not use a fixed seed across the cluster rather than a seed dependent on it's rank ?

Comment (Cpruce) [reply PyTorch/Segmentation/MaskRCNN/pytorch/train_net_herring.py:246]: @karan6181 yes. @piyushghai The user can set the per rank seeds to whatever he/she wants if we just pass the seed argument through to the seed functions.

Comment (karan6181) [reply PyTorch/Segmentation/MaskRCNN/pytorch/train_net_herring.py:246]: From what I have observed, TF and MX uses a same seed value across the cluster. However, only PT uses the seed based on ranks. I am also inclined towards using the same seed value across the cluster unless someone have a conflict reason!!

Comment (karan6181) [reply PyTorch/Segmentation/MaskRCNN/pytorch/train_net_herring.py:246]: This is how NVIDIA forks uses seed in PyTorch:
https://github.com/HerringForks/DeepLearningExamples/blob/master/PyTorch/LanguageModeling/BERT/run_pretraining.py#L499
https://github.com/HerringForks/DeepLearningExamples/blob/master/PyTorch/Detection/SSD/main.py#L151

In both cases, they use ranks

File: TensorFlow2/Segmentation/MaskRCNN/mask_rcnn_main.py (hunk 2 above)
Comment (karan6181) [reply TensorFlow2/Segmentation/MaskRCNN/mask_rcnn_main.py]: I have revert the TF changes for now. Currently focusing on PT models.

**Review by piyushghai — APPROVED** [review]
